lib/pymedphys/tests/mosaiq/test_mimicked_db.py

```diff
@@ -0,0 +1,144 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pymedphys._imports import numpy as np
+from pymedphys._imports import pytest
+
+import pymedphys
+import pymedphys.beta
+from pymedphys._mosaiq import helpers
+
+from . import _connect
+from .data import mimics
+
+PATIENT_ID = 989898
+FIELD_ID = 88043
+A_TREATMENT_DATETIME = "2020-04-27 08:03:28.513"
+MACHINE_ID = "2619"
+FIELD_NAME = "3ABUT"
+TIMEZONE = "Australia/Sydney"
+FIRST_NAME = "MOCK"
+LAST_NAME = "PHYSICS"
+FULL_NAME = f"{LAST_NAME}, {FIRST_NAME.capitalize()}"
+QCL_LOCATION = "Physics_Check"
+AN_UNCOMPLETED_QCL_DUE_DATETIME = "2021-05-21 23:59:59"
+QCL_COMPLETED_DATETIMES = ["2021-04-14 09:11:30.387", "2021-04-14 09:11:35.383"]
+
+
+@pytest.fixture(name="connection")
+def connection_base():
+    """ will create the test database, if it does not already exist on the instance """
+    mimics.create_db_with_tables()
+    return _connect.connect(database=mimics.DATABASE)
+
+
+@pytest.fixture(name="trf_filepath")
+def trf_filepath_base():
+    data_paths = pymedphys.zip_data_paths("metersetmap-gui-e2e-data.zip")
+    date = A_TREATMENT_DATETIME.split(" ")[0]
+
+    filtered_paths = [
+        path
+        for path in data_paths
+        if date in str(path) and str(FIELD_ID) in str(path) and path.suffix == ".trf"
+    ]
+
+    return filtered_paths[0]
+
+
+@pytest.fixture(name="dicom_filepath")
+def dicom_filepath_base():
+    data_paths = pymedphys.zip_data_paths("metersetmap-gui-e2e-data.zip")
+
+    filtered_paths = [
+        path for path in data_paths if path.name == f"{PATIENT_ID}_{FIELD_NAME}.dcm"
+    ]
+
+    return filtered_paths[0]
+
+
+@pytest.mark.mosaiqdb
+def test_get_patient_name(connection):
+    name = helpers.get_patient_name(connection, PATIENT_ID)
+    assert name == FULL_NAME
+
+
+@pytest.mark.mosaiqdb
+def test_get_patient_fields(connection):
+    tx_fields = helpers.get_patient_fields(connection, PATIENT_ID)
+    field_id = tx_fields["field_id"].iloc[0]
+    assert field_id == FIELD_ID
+
+
+@pytest.mark.mosaiqdb
+def test_get_treatment_times(connection):
+    treatment_times = helpers.get_treatment_times(connection, FIELD_ID)
+    assert np.datetime64(A_TREATMENT_DATETIME) in treatment_times["start"].tolist()
+
+
+@pytest.mark.mosaiqdb
+def test_get_treatments(connection):
+    time_delta = np.timedelta64(4, "h")
+    start = np.datetime64(A_TREATMENT_DATETIME) - time_delta
+    end = np.datetime64(A_TREATMENT_DATETIME) + time_delta
+
+    treatments = helpers.get_treatments(connection, start, end, MACHINE_ID)
+    assert np.datetime64(A_TREATMENT_DATETIME) in treatments["start"].tolist()
+
+
+@pytest.mark.mosaiqdb
+def test_trf_identification(connection: pymedphys.mosaiq.Connection, trf_filepath):
+    delivery_details = pymedphys.beta.trf.identify(connection, trf_filepath, TIMEZONE)
+    assert delivery_details.field_id == FIELD_ID
+    assert delivery_details.first_name == FIRST_NAME
```

Comment:
Is capitalization important?

Reply:
See above.

Reply:
Fixed in 43edbc618acd60f55eae27d30944342b18b9fb4f